ui/tests/test_ui.py

```diff
@@ -0,0 +1,169 @@
+import os
+from playwright.sync_api import sync_playwright, Page
+
+
+def test_ui_start():
+    with sync_playwright() as ctx:
+        browser = ctx.chromium.launch(headless=True)
+        page = browser.new_page()
+        page.goto("http://localhost:8080")
+        page.wait_for_load_state("load")
+        assert page.title() == "Kubeseal Webgui"
+        browser.close()
+
+
+def test_secret_form_with_value():
+    with sync_playwright() as ctx:
+        browser = ctx.chromium.launch(headless=True)
+        page = browser.new_page()
+        page.goto("http://localhost:8080")
+        page.wait_for_load_state("load")
+
+        disabled_encrypt_button(page)
+        namespace_select(page)
+        secret_name(page)
+        scope_strict(page)
+        add_secret_key_value(page)
+        click_encrypt_button(page)
+
+        browser.close()
+
+
+def test_secret_form_with_file():
+    with sync_playwright() as ctx:
+        browser = ctx.chromium.launch(headless=True)
+        page = browser.new_page()
+        page.goto("http://localhost:8080")
+        page.wait_for_load_state("load")
+
+        disabled_encrypt_button(page)
+        namespace_select(page)
+        secret_name(page)
+        scope_strict(page)
+        add_secret_key_file(page)
+        click_encrypt_button(page)
+
+        browser.close()
+
+
+def test_secret_form_with_invalid_file():
+    with sync_playwright() as ctx:
+        browser = ctx.chromium.launch(headless=True)
+        page = browser.new_page()
+        page.goto("http://localhost:8080")
+        page.wait_for_load_state("load")
+
+        file_input = page.locator('input[type="file"]#input-16')
+        # Try to upload a non-valid file
+        invalid_file_path = os.path.join(os.getcwd(), "large_test_file.txt")
```

Comment:
When we are inside of pytest, we could use it to hand a tmpfile into the test function and use that instead of the current working directory.